### ISSUE TITLE
[Bug]: (Crash) Async arrow function with same parameter name under noNewArrow assumption

### ISSUE THREAD

**Issue by p51lee** [babel/babel#15170]
### 💻

- [ ] Would you like to work on a fix?

### How are you using Babel?

Other (Next.js, Gatsby, vue-cli, ...)

### Input code

```js
x = async x => 0 ;
```

### Current and expected behavior

`input.js` is terminated normally:
```console
$ node input.js # no error
```
but under `noNewArrow` assumption, Babel fails to transpile `input.js`:

By the way, when I used variable declaration instead of assignment expression, (`var x = async x => 0`)  the transpilation terminates normally.

### Environment

- Babel version: 7.19.1
- Node: v18.11.0
- npm version: 8.19.2
- OS: macos 12.4

**Comment by liuxingbaoyu** [comment]
I tried to reproduce this without success.😕
![image](https://user-images.githubusercontent.com/30521560/200882297-4e56589f-7eb5-4ce8-9bae-9bdc44244fd9.png)

**Comment by p51lee** [comment]
Thank you for your answer.

I think there might be a problem with the web REPL. When I applied the noNewArrows assumption in the REPL of the homepage, nothing changed, so I ran it in my local environment.

Currently, even a simple example of https://github.com/babel/babel/issues/15155
```js
var f1 = () => {};
new f1();  
```
doesn't seem to work on the web.

<img width="1498" alt="image" src="https://user-images.githubusercontent.com/68288688/201003189-58378e90-457a-4a0e-9e0a-e298e32ddd00.png">

However, if I apply the assumption in the local environment, it works fine.
